src/NUnitCore/core/Compatibility.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NUnit.Core
+{
+    public static class Compatibility
+    {
+        private const string WORK_FILE = "COMPATIBILITY.TXT";
+        private const string REPORT_FORMAT = "{0}:{1}:{2}";
+        private const BindingFlags ALL_MEMBERS = 
+            BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic;
+
+        private static string _workFilePath;
+        private static StreamWriter _writer;
+
+        #region Initialization and Cleanup
+
+        /// <summary>
+        /// This method initializes the working directory and file path
+        /// for compatibility reporting without starting collection.
+        /// It should only be called by the console runner itself in
+        /// the primary AppDomain.
+        /// </summary>
+        public static void Initialize(string workDirectory)
+        {
+            _workFilePath = Path.Combine(workDirectory, WORK_FILE);
+            File.Create(_workFilePath);
+        }
+
+        /// <summary>
+        /// This method is called once in each test AppDomain to
+        /// begin collecting data.
+        /// </summary>
+        /// <param name="workDirectory"></param>
+        public static void BeginCollection(string workDirectory)
+        {
+            _workFilePath = Path.Combine(workDirectory, WORK_FILE);
+
+            _writer = new StreamWriter(_workFilePath, true)
+            {
+                AutoFlush = true
+            };
+        }
+
+        /// <summary>
+        /// This method is called in each test AppDomain after all
+        /// collection is complete.
+        /// </summary>
+        public static void EndCollection()
+        {
+            if (_writer != null)
+            {
+                _writer.Close();
+                _writer = null;
```

Comment:
Setting `_writer` to null doesn't have any observable effect as compared to not setting it to null, does it?